File: abc_training/D_diff/D_Stamp.py
#               D - Stamp
# ----------------------------------------
# 問題
# https://atcoder.jp/contests/abc185/tasks/abc185_d

# こういう問題を速解きできるように

# AC
# ----------------------------------------
from math import ceil

n, m = map(int, input().split())
A = [-1] + [int(a) - 1 for a in input().split()]

A.sort()
A.append(n)

diff = [A[i+1] - A[i] - 1 for i in  range(len(A)-1) if A[i+1] - A[i] > 1]

if diff:
    stamp_size = min(diff)
    res = sum( ceil(seg / stamp_size)  for seg in diff )
    print(res)
else:
    print(0)

File: merge_contests_atcoder_training/abc185/src/bin/D_Stamp.cpp
//              D - Stamp
// ----------------------------------------
// 問題
// https://atcoder.jp/contests/abc185/tasks/abc185_d
// ----------------------------------------

#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
#define rep(i, n) for (int i = 0; i < (int)(n); i++)
#define ALL(A) A.begin(), A.end()

template <typename T>
void print_vector(vector<T>& vec) {
  cerr << "[ ";
  for (int i = 0; i < vec.size(); i++) {
    if (i < vec.size() - 1) cerr << vec.at(i) << " ";
    else cerr << vec.at(i);
  }
  cerr << " ]" << endl;
}

int main() {
    ll n, m;
    cin >> n >> m;
    vector<ll> A(m+2, 0), diff(m);
    rep(i, m) {
        int a; cin >> a;
        A[i+1] = a - 1;
    }
    sort(ALL(A));
    

    rep(i, m) {
        diff[i] = A[i+1] - A[i];
    }

    print_vector(A);
    print_vector(diff);
}